addons/account/models/account_move.py

```diff
@@ -2554,3581 +2323,1276 @@ def _get_invoice_computed_reference(self):
-    full_reconcile_id = fields.Many2one('account.full.reconcile', string="Matching", copy=False, index='btree_not_null', readonly=True)
-    matched_debit_ids = fields.One2many('account.partial.reconcile', 'credit_move_id', string='Matched Debits',
-        help='Debit journal items that are matched with this journal item.', readonly=True)
-    matched_credit_ids = fields.One2many('account.partial.reconcile', 'debit_move_id', string='Matched Credits',
-        help='Credit journal items that are matched with this journal item.', readonly=True)
-    matching_number = fields.Char(string="Matching #", compute='_compute_matching_number', store=True, help="Matching number for this line, 'P' if it is only partially reconcile, or the name of the full reconcile if it exists.")
-    is_account_reconcile = fields.Boolean(related='account_id.reconcile', string='Account Reconcile', help='Technical field for the matching link widget')
-
-    # ==== Analytic fields ====
-    analytic_line_ids = fields.One2many('account.analytic.line', 'move_id', string='Analytic lines')
-    analytic_account_id = fields.Many2one('account.analytic.account', string='Analytic Account',
-        index='btree_not_null', compute="_compute_analytic_account_id", store=True, readonly=False, check_company=True, copy=True)
-    analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Analytic Tags',
-        compute="_compute_analytic_tag_ids", store=True, readonly=False, check_company=True, copy=True)
-
-    # ==== Onchange / display purpose fields ====
-    recompute_tax_line = fields.Boolean(store=False, readonly=True,
-        help="Technical field used to know on which lines the taxes must be recomputed.")
-    display_type = fields.Selection([
-        ('line_section', 'Section'),
-        ('line_note', 'Note'),
-    ], default=False, help="Technical field for UX purpose.")
-    is_rounding_line = fields.Boolean(help="Technical field used to retrieve the cash rounding line.")
-    exclude_from_invoice_tab = fields.Boolean(help="Technical field used to exclude some lines from the invoice_line_ids tab in the form view.")
-
-    _sql_constraints = [
-        (
-            'check_credit_debit',
-            'CHECK(credit * debit=0)',
-            'Wrong credit or debit value in accounting entry !'
-        ),
-        (
-            'check_accountable_required_fields',
-             "CHECK(COALESCE(display_type IN ('line_section', 'line_note'), 'f') OR account_id IS NOT NULL)",
-             "Missing required account on accountable invoice line."
-        ),
-        (
-            'check_non_accountable_fields_null',
-             "CHECK(display_type NOT IN ('line_section', 'line_note') OR (amount_currency = 0 AND debit = 0 AND credit = 0 AND account_id IS NULL))",
-             "Forbidden unit price, account and quantity on non-accountable invoice line"
-        ),
-        (
-            'check_amount_currency_balance_sign',
```

Comment:
Is this constraint removed? It seems important to keep it in SQL.

Reply:
Added back, was removed while trying a new model but forgot to add it back